features/mbedtls/targets/TARGET_Silicon_Labs/aes_aes.c

```diff
@@ -0,0 +1,498 @@
+/*
+ *  Hardware-accelerated AES implementation for Silicon Labs devices
+ *  containing an AES peripheral.
+ *
+ *  Copyright 2017, Silicon Laboratories, Inc.
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "mbedtls/aes.h"
+
+#if defined(MBEDTLS_AES_C)
+#if defined(MBEDTLS_AES_ALT)
+
+#include "em_device.h"
+
+#if defined(AES_PRESENT) && (AES_COUNT == 1)
+#include "em_aes.h"
+#include "em_cmu.h"
+#include "em_bus.h"
+#include <string.h>
+
+#if defined(MBEDTLS_THREADING_C)
```

Comment:
We're shipping the same codebase to a broader audience then just mbed OS. Keep in mind these are mbed TLS drivers, and not dependent on mbed OS.
If MBEDTLS_THREADING_C gets defined somehow, then that would indicate threading support is present, and compilation won't fail because the threading support is there, as indicated by the compilation flag.